datafusion/expr/src/expr.rs

```diff
@@ -580,39 +591,24 @@ impl Not for Expr {
 
     fn not(self) -> Self::Output {
         match self {
-            Expr::Like {
-                negated,
-                expr,
-                pattern,
-                escape_char,
-            } => Expr::Like {
-                negated: !negated,
-                expr,
-                pattern,
-                escape_char,
-            },
-            Expr::ILike {
-                negated,
-                expr,
-                pattern,
-                escape_char,
-            } => Expr::ILike {
-                negated: !negated,
-                expr,
-                pattern,
-                escape_char,
-            },
-            Expr::SimilarTo {
-                negated,
-                expr,
-                pattern,
-                escape_char,
-            } => Expr::SimilarTo {
-                negated: !negated,
-                expr,
-                pattern,
-                escape_char,
-            },
+            Expr::Like(like) => Expr::Like(Like::new(
+                !like.negated,
+                like.expr,
+                like.pattern,
+                like.escape_char,
+            )),
```

Comment:
Another way you can write this type of match is like 

```suggestion
            Expr::Like(Like { negated, expr, pattern, escape_char} ) => Expr::Like(Like::new(
                negated,
                expr,
                pattern,
                escape_char,
            )),
```

The benefit of doing so is that if any new field is added to `Like` the compiler will force you to update all places that use `Like` to add the new field where as this way you would have to find them manually